tests/e2e/test_api.py

```diff
@@ -822,37 +825,120 @@ def test_transcripts_storing_standalone():
+    """
+
+    def filter_logs(logs: str, last_log_line: str) -> str:
+        filtered_logs = []
+        new_logs = False
+        for line in logs.split("\n"):
+            if line == last_log_line:
+                new_logs = True
+                continue
+            if new_logs:
+                filtered_logs.append(line)
+        return "\n".join(filtered_logs)
+
+    def get_last_log_line(logs: str) -> str:
+        return [line for line in logs.split("\n") if line][-1]
+
+    # constants from tests/config/cluster_install/ols_manifests.yaml
+    data_collection_container_name = "ols-sidecar-user-data-collector"
     pod_name = cluster_utils.get_single_existing_pod_name()
 
-    # there are multiple tests running agains cluster, so transcripts
+    # there are multiple tests running agains cluster, so user data
     # can be already present - we need to ensure the storage is empty
     # for this test
-    transcripts = cluster_utils.list_path(pod_name, transcripts_path)
-    if transcripts:
-        cluster_utils.remove_dir(pod_name, transcripts_path)
-        assert cluster_utils.list_path(pod_name, transcripts_path) == []
+    user_data = cluster_utils.list_path(pod_name, OLS_USER_DATA_PATH)
+    if user_data:
+        cluster_utils.remove_dir(pod_name, OLS_USER_DATA_PATH + "/feedback")
+        cluster_utils.remove_dir(pod_name, OLS_USER_DATA_PATH + "/transcripts")
+        assert cluster_utils.list_path(pod_name, OLS_USER_DATA_PATH) == []
+
+    # data shoud be pruned now and this is the point from which we want
+    # to check the logs
+    container_log = cluster_utils.get_container_log(
+        pod_name, data_collection_container_name
+    )
+    last_log_line = get_last_log_line(container_log)
 
-    response = client.post(
-        "/v1/query",
-        json={
-            "query": "what is kubernetes?",
-        },
-        timeout=LLM_REST_API_TIMEOUT,
+    # wait the collection period for some extra to give the script a
+    # chance to log what we want to check
+    time.sleep(OLS_USER_DATA_COLLECTION_INTERVAL + 1)
+
+    # we just check that there are no data and the script is working
+    container_log = cluster_utils.get_container_log(
+        pod_name, data_collection_container_name
     )
-    assert response.status_code == requests.codes.ok
+    logs = filter_logs(container_log, last_log_line)
+
+    assert "collected" not in logs
+    assert "data uploaded with request_id:" not in logs
+    assert "uploaded data removed" not in logs
+    assert "data upload failed with response:" not in logs
+    assert "contains no data, nothing to do..." in logs
 
-    transcript = cluster_utils.get_single_existing_transcript(
-        pod_name, transcripts_path
+    # get the log point for the next check
+    last_log_line = get_last_log_line(container_log)
+
+    # create a new data
+    cluster_utils.create_file(
+        pod_name, OLS_USER_DATA_PATH + "/feedback/test-file.json", '{"testy": "test"}'
```

Comment:
would be better to actually submit a feedback request to OLS to get this generated, any reason not to?

Reply:
I found that unnecessary. With this, the test doesn't depend on any implementation from the ols service itself.
Obviously, the collector is useless without the service, but I like its independence.

Reply:
the benefit from submitting a feedback request is that it makes it a true e2e test, confirming that the collector is able to collect exactly the files (name, type, content, etc) created by the feedback endpoint.

by creating the file manually, you're introducing a potential disconnect between what is being tested(collector can collect a specific file created by the test), and what the system actually does at runtime(collector collects a file created by the feedback endpoint).  It becomes more of an integration test than a true e2e test.

Reply:
You are correct. Updated.